script.js

```diff
@@ -154,6 +154,55 @@ buttons.forEach((button, index) => {
   });
 });
 
+// 1.0 LOCAL STORAGE
+//   1.1 Testing for availability
+function storageAvailable(type) {
+  let storage;
+  try {
+    storage = window[type];
+    const x = "__storage_test__";
+    storage.setItem(x, x);
+    storage.removeItem(x);
+    return true;
+  } catch (error) {
+    return (
+      error instanceof DOMException &&
+      // everything except Firefox
+      (error.code === 22 ||
+        // Firefox
+        error.code === 1014 ||
+        // test name field too, because code might not be present
+        // everything except Firefox
+        error.name === "QuotaExceededError" ||
+        // Firefox
+        error.name === "NS_ERROR_DOM_QUOTA_REACHED") &&
+      // acknowledge QuotaExceededError only if there's something already stored
+      storage &&
+      storage.length !== 0
+    );
+  }
+}
+
+//   1.2 savingData
+function saveData() {
+  if (storageAvailable("localStorage")) {
+    let nameInput = document.getElementById("name-input");
+    let emailInput = document.getElementById("email-input");
+    let messageInput = document.getElementById("message-box");
+    let formObj = {
+      name: nameInput.value,
+      email: emailInput.value,
+      message: messageInput.value,
+    };
+
+    for (let i = 0; i < Object.keys(formObj).length; i++) {
+      localStorage.setItem(Object.keys(formObj)[i], Object.values(formObj)[i]);
+    }
```

Comment:
Great work so far , but I would suggest that you use only one object in localStorage instead of creating an item for each property this will reduce calls to set localStorage item and this was clearly mentioned in the requirements 
project requirements:
You must create a single JavaScript object with all the data from the entire form and save it in local storage. Do not create one local storage entry per input field.